Build error in project-to-project reference scenario after reacting to the buildOptions changes

## Steps to reproduce

```
git clone https://github.com/jeremymeng/project-ref-test.git
cd project-ref-test
dotnet restore
cd UnitTests.NetCore\ClassLibrary2
dotnet build
```

## Expected  behavior

Build succeeds.

## Actual behavior

Project `ClassLibrary2`, which references project `ClassLibrary1`, failed to build.

```
Project ClassLibrary1 (.NETStandard,Version=v1.3) will be compiled because expected outputs are missing
Compiling ClassLibrary1 for .NETStandard,Version=v1.3

Compilation succeeded.
    0 Warning(s)
    0 Error(s)

Time elapsed 00:00:01.8021542
Project ClassLibrary2 (.NETStandard,Version=v1.3) will be compiled because dependencies changed
Compiling ClassLibrary2 for .NETStandard,Version=v1.3
C:\Program Files\dotnet\dotnet.exe compile-csc @E:\github\jeremymeng\project-ref-test\UnitTests.NetCore\ClassLibrary2\obj\Debug\netstandard1.3\dotnet-compile.rsp returned Exit Code 1
E:\github\jeremymeng\project-ref-test\UnitTests\Library\ClassLibrary2\Bar.cs(5,7): error CS0246: The type or namespace name 'ClassLibrary1' could not be found (are you missing a using directive or an assembly reference?)
E:\github\jeremymeng\project-ref-test\UnitTests\Library\ClassLibrary2\Bar.cs(20,21): error CS0246: The type or namespace name 'Foo' could not be found (are you missing a using directive or an assembly reference?)
E:\github\jeremymeng\project-ref-test\UnitTests\Library\ClassLibrary2\Bar.cs(15,26): error CS0246: The type or namespace name 'IFoo' could not be found (are you missing a using directive or an assembly reference?)

Compilation failed.
    0 Warning(s)
    3 Error(s)

Time elapsed 00:00:01.1818135
```

### Notes

* The signing doesn't matter: same result after removing `keyFile` lines.

* The dependency `ClassLibrary1` seems to be found correctly, and was built first successfully as shown above.

* Before reacting to the changes, the build succeeded, with the warnings:

```
git checkout HEAD~1
dotnet build

Project ClassLibrary1 (.NETStandard,Version=v1.3) will be compiled because inputs were modified
Compiling ClassLibrary1 for .NETStandard,Version=v1.3
E:\github\jeremymeng\project-ref-test\UnitTests.NetCore\ClassLibrary1\project.json(4,15): warning DOTNET1015: The 'compile' option is deprecated. Use 'compile' in 'buildOptions' instead.
E:\github\jeremymeng\project-ref-test\UnitTests.NetCore\ClassLibrary1\project.json(5,26): warning DOTNET1015: The 'compilationOptions' option is deprecated. Use 'buildOptions' instead.
E:\github\jeremymeng\project-ref-test\UnitTests.NetCore\ClassLibrary1\project.json(13,30): warning DOTNET1015: The 'compilationOptions' option is deprecated. Use 'buildOptions' instead.

Compilation succeeded.
    3 Warning(s)
    0 Error(s)

Time elapsed 00:00:01.6146769
Project ClassLibrary2 (.NETStandard,Version=v1.3) will be compiled because dependencies changed
Compiling ClassLibrary2 for .NETStandard,Version=v1.3
E:\github\jeremymeng\project-ref-test\UnitTests.NetCore\ClassLibrary2\project.json(4,15): warning DOTNET1015: The 'compile' option is deprecated. Use 'compile' in 'buildOptions' instead.
E:\github\jeremymeng\project-ref-test\UnitTests.NetCore\ClassLibrary2\project.json(5,26): warning DOTNET1015: The 'compilationOptions' option is deprecated. Use 'buildOptions' instead.
E:\github\jeremymeng\project-ref-test\UnitTests.NetCore\ClassLibrary2\project.json(13,30): warning DOTNET1015: The 'compilationOptions' option is deprecated. Use 'buildOptions' instead.

Compilation succeeded.
    3 Warning(s)
    0 Error(s)

Time elapsed 00:00:01.5832139
```
## Environment data
`dotnet --info` output:

--- a/src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs
+++ b/src/Microsoft.DotNet.ProjectModel/Compilation/LibraryExporter.cs
@@ -299,7 +299,7 @@ namespace Microsoft.DotNet.ProjectModel.Compilation
                     builder.AddRuntimeAsset(new LibraryAsset(Path.GetFileName(pdbPath), Path.GetFileName(pdbPath), pdbPath));
                 }
             }
-            else if (HasSourceFiles(project))
+            else if (HasSourceFiles(project, compilerOptions))
             {
                 var outputPaths = project.GetOutputPaths(_buildBasePath, _solutionRootPath, _configuration, _runtime);
 
@@ -337,10 +337,8 @@ namespace Microsoft.DotNet.ProjectModel.Compilation
             return builder.Build();
         }
 
-        private bool HasSourceFiles(ProjectDescription project)
+        private bool HasSourceFiles(ProjectDescription project, CommonCompilerOptions compilerOptions)
         {
-            var compilerOptions = project.TargetFrameworkInfo.CompilerOptions;
-
             if (compilerOptions.CompileInclude == null)
             {
                 return project.Project.Files.SourceFiles.Any();

